### ISSUE TITLE
Tests failing

### ISSUE THREAD

**Issue by artfulrobot** [eileenmcnaughton/org.wikimedia.geocoder#51]
On a clean install the extension will not install due to a [bug in the SQL file](https://github.com/eileenmcnaughton/org.wikimedia.geocoder/blob/master/sql/install.sql#L15). After that, the tests mostly error due to geocoders not being called (so no `geo_code_1` type fields are returned by api3, e.g. in `testOpenStreetMapsFailsFallsBackToUSLookup`).

**Comment by heitzjm** [comment]
I've just found out why tests were failing : it is related to : 
https://github.com/eileenmcnaughton/org.wikimedia.geocoder/blob/7342cdd1001b182113724da0bacc8162b04ac745/CRM/Utils/Geocode/Geocoder.php#L102

Data in some metaproviders contains several iso codes for countries, while we compare them with numeric country ids. Thus is doesn't work.

I noticed a further problem with the tests which involves two sections of code :  

https://github.com/eileenmcnaughton/org.wikimedia.geocoder/blob/7342cdd1001b182113724da0bacc8162b04ac745/tests/phpunit/GeocoderTest.php#L212-L216

https://github.com/eileenmcnaughton/org.wikimedia.geocoder/blob/7342cdd1001b182113724da0bacc8162b04ac745/sql/open_postcode_geo-test.sql#L3-L11

It seems that I might not have the `civicrm_open_postcode_geo_uk` table in the database (not fully sure about it, because of all the tests I run), while I had the `civicrm_geonames_lookup` in the database. I changed to test the existence of `civicrm_open_postcode_geo_uk`, but it didn't solve the problem : I still had a problem with transactions in the database : it seems that data where inserted in one transaction (maybe unfinished or rolled back) while data was search in another transaction. Adding START TRANSACTION and COMMIT in the data file solved the problem.

In the end, I was able to successfully run all the tests.

Now, I see two ways to solve the problems : 
- correct the IDs of the countries in the provider Data 
- write code to check whether a country id is numeric or not, and get the country id if it is not numeric by assuming it is a country isocode.

I guess the second will be more effective as some meta providers are designed for not beeing updated.
